www/guides/react.md

```diff
@@ -0,0 +1,371 @@
++ ReactDOM.render(
++    <React.StrictMode>
++      <App />
++    </React.StrictMode>,
++    document.getElementById('root'),
++  );
+```
+
+> 💡 Tip: [Strict Mode](https://reactjs.org/docs/strict-mode.html) is a tool for highlighting potential problems in React code.
+
+You shouldn't need to restart Snowpack to see this, it should look like this:
+
+Image: screenshot of the project, showing text saying "edit `src/App.jsx` and save to reload," text that says "Page has been open for" and the number of seconds then "seconds". A link with the text Learn React that links to the React website.
+
+## Styling your Snowpack/React project
+
+When you add assets like images or CSS, Snowpack includes them in your final build. If you already know React, this process should look pretty familiar.
+
+> 💡 Tip: As you're doing this, you should not need to reload the page or restart Snowpack. Snowpack automatically updates the project in the browser as you edit code.
+
+Add this file `logo.svg` (link to file) to your `src` directory. Now you can import it into your `App.jsx` and use it in an `img` tag to display it.
+
+```diff
+import React, { useState, useEffect } from 'react';
++ import logo from './logo.svg';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
++        <img src={logo} className="App-logo" alt="logo" />
+        <p>
+```
+
+Image: the React logo (a blue atom) is now at the top of the page
+
+The project already has index.css for global styles. For CSS that's only for a specific component, a common design pattern is to add it in a CSS file with the same base name as the component. So the file for `App.jsx` would be `App.css` with this pattern.
+
+> 💡 Tip: Snowpack has built-in support for [CSS Modules](https://www.snowpack.dev/#import-css-modules) and if you'd like to use SASS there is an official [SASS Plugin](https://www.snowpack.dev/#sass).
+
+Create `src/App.css` and add this CSS:
+
+```css
+.App {
+  text-align: center;
+}
+.App code {
+  background: #fff3;
+  padding: 4px 8px;
+  border-radius: 4px;
+}
+.App p {
+  margin: 0.4rem;
+}
+.App-logo {
+  height: 40vmin;
+  pointer-events: none;
+}
+
+@media (prefers-reduced-motion: no-preference) {
+  .App-logo {
+    animation: App-logo-spin infinite 20s linear;
+  }
+}
+
+.App-header {
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
+
+.App-link {
+  color: #61dafb;
+}
+
+@keyframes App-logo-spin {
+  from {
+    transform: rotate(0deg);
```

Comment:
This affects nothing at all, but you can omit the `from [` block here as it’s inferred, to save lines